designs/virtualization/4.x/create-vm-dropdowns/create-vm-dropdowns.md

```diff
@@ -0,0 +1,111 @@
+- Adding descriptions to the dropdown fields (using this pattern https://github.com/patternfly/patternfly-design-kit/issues/218)
+- Renaming fields
+- Reordering fields
+
+
+![Create VM - General-default](img/1-0.jpg)
+Landing on the General step, Default mode.
+Section titles added, and fields reorgenized and renamed.
+'General' section:
+- 'Name'
+- 'Description'
+'Resources & configuration' section'
+- 'Operating system'
+- 'VM configuration' (previously Workload profile)
+- 'Flavor (CPU and Memory)' (previously only 'flavor')
+'Operating system image source' section
+- 'Boot operating system via' (previously 'source')
+
+## General step
+
+![Create VM - General](img/1-1.jpg)
+On selecting an Operating system, the VM configuration and Flavor (CPU and Memory) fields will be enabled and selected on default
+
+![Create VM - General - VM configuration](img/1-1-1.jpg)
+VM Configuration options
+Server is the default
+
+![Create VM - General - flavor](img/1-1-2.jpg)
+Flavor options
+Medium is the default
+
+![Create VM - General](img/1-2-1.jpg)
+'Boot operating system via' field options
+
+![Create VM - General](img/1-2.jpg)
+Selecting an option form the 'Boot operating system via' field will expose below all the fields required to connect the VM to an OS image source.
+For example: on selecting the 'Existing PVC' option, additional fields will appear below, required to select a PVC available on the cluster.
+The disk name should be added here as well to establish a better link between this step and the Storage step.
+An infobox will notify the user that a disk will be added to to the disks list on the Storage step.
+
+![Create VM - OS disk on list](img/1-3.jpg)
+Within the storage step, the disk added on the general step will appear as the first item on the list.
+This disk cannot be deleted but can be modified.
+
+![Create VM - OS disk modal](img/1-3-1.jpg)
+Editing that disk will display a warning notification within the edit modal.
+
+## Netwroking step
+
+![Create VM - Networking -list](img/2-0.jpg)
+NIC list.
+In case no PXE boot the OS image selected on the General step, a default NIC will be added.
+This NIC  exposes the VM to the cluster.
+
+![Create VM - Networking -Default](img/2-1.jpg)
+On editing this default NIC, a warning message will remind the user that modifying this NIC might break that 'expoing to cluster' configuration.
+
+![Create VM - Networking -Add](img/2-2.jpg)
+Add NIC modal.
+The Advanced drawer is closed by default, as seen in the previous image.
+All fields had moved into the advanced drawer, except the Name and Network fields.
+Network field does not require description for each option since these options are created and managed by the user.
+
+![Create VM - Networking -Model](img/2-3.jpg)
+'Model' dropdown options
+
+![Create VM - Networking -Binding](img/2-4.jpg)
+'Binding method' dropdown options (previously 'Type')
+
+## Storage
+
+![Create VM - Storage -list](img/3-0.jpg)
```

Comment:
What if no disk has been added we say in the description text a blank disk has been added to ensure the VM is operational. Otherwise we don't say that.

Reply:
Because this is circumstantial, I rather say this text when we do create a default Disk, within the disk modal itself (see image 3-0-0). This is the same thing we did with the NIC created due to PXE boot option on the General step.

I did change the description in the Disk list to be:
"Manage virtual machine disks. Disks are backed by PVCs which can be seen in storage > persistent volume claims
At least one disk is required to make the virtual machine operational"
But now that I think about it we should not prevent the user from deleting the last disk, but just warn him in an empty state that at least one disk is required, and drop the last line from the description.

Reply:
Let's pull someone like @jelkosz or @suomiy in for their opinion. I believe we've always forced at least one disk. 
I'm not sure what advantage we are giving the user if we allow them to delete it but then require them to add it back. 
I guess for most cases though we shouldn't see users who aren't aware of the storage they are setting for a VM. 
Is a blank disk really needed for real world examples or just to ensure when someone randomly creates a VM that they have a disk?

Reply:
>"Manage virtual machine disks. Disks are backed by PVCs which can be seen in storage > persistent volume claims

That is true, but just for some of them. Some disks are backed by Secret, ConfigMap or by nothing (Container, CloudInit). Maybe we should also mention DVs when mentioning PVCs (also not applicable to all of them eg when you attach PVC). Also some PVCs will not be visible when you create for example the template, but some PVCs will be created when you create a template.

Stating a generic truth here can be pretty complicated.

>At least one disk is required to make the virtual machine operational"
But now that I think about it we should not prevent the user from deleting the last disk, but just warn him in an empty state that at least one disk is required, and drop the last line from the description.

Yes we should not allow deleting source rootdisk, but other than that I think it is best not to add other restrictions. These might not be much useful when creating for example templates. Maybe the user wants just PXE template and wants to be adding disks later when creating VMs.

Reply:
@suomiy This relates to the previous comment. Does it make sense to offer to create a disk from a container within the storage step? Is it really equivalent to a "disk" in virtualization?
I am trying to confirm or deny my assumption that a "Container disk" should be available only for select OS image source

Reply:
Yes it makes sense. The container might include many things, not just bootable image.

But it is not equivalent  to a "disk" in traditional virtualization as it is not a persistent storage

Reply:
@suomiy So:
1. Uploading and OS image from a container does create a disk even though it does not create PVC
2. We do need to offer the option of creating a disk within the storage step, with uploading content form a container (creates PVC)
is that correct?

Reply:
IMO uploading a container does not make sense. You should just use a relevant registry in your environment.

Container should be just an image string/

Uploading itself will take an image and create DV/PVC and upload it to that.

Reply:
Ok, I have removed the "Disks are backed by PVCs which can be seen in storage > persistent volume claims" description from the top of the storage step.

Removing the OS source image disk from the equation for a second: If we allow from the Storage step to create a disk (like container), that does not involve PVC then this description is not true anymore